### PR TITLE
Feature/custom slugs

### PR DESCRIPTION
**Pull request by krisquigley**
Adds slug model and associations

### CODE REVIEW

**Comment by ryantownsend** [comment]
👍  LGTM.

File: lib/flex_commerce_api/json_api_client_extension/requestor.rb [hunk 1]
```diff
@@ -35,10 +35,9 @@ def resource_path(parameters, record = nil)
       end
 
       def encoded(part)
+        # Do not encode the slug
```
Comment (garytaylor) [inline lib/flex_commerce_api/json_api_client_extension/requestor.rb]: Should this comment still be here ?  There doesnt appear to be any code backing it up ?

Comment (krisquigley) [inline lib/flex_commerce_api/json_api_client_extension/requestor.rb]: Sorry, @garytaylor, thought I had just removed this comment.

**Comment by kmcd** [comment]
Looks good 👍

**Pull request merged**